File: backend/core-api/src/main/java/ai/govbiz/core/domain/support/SupportProgram.java
package ai.govbiz.core.domain.support;

import java.time.LocalDate;
import java.util.List;

public record SupportProgram(
        String id,
        String title,
        String organization,
        String summary,
        List<String> categories,
        List<String> regions,
        String targetDescription,
        String supportAmount,
        String applicationPeriod,
        LocalDate applicationStartDate,
        LocalDate applicationEndDate,
        SupportProgramStatus status,
        String sourceName,
        String sourceUrl,
        List<String> matchedReasons
) {
}

File: backend/core-api/src/main/kotlin/ai/govbiz/core/domain/support/SupportProgram.kt
package ai.govbiz.core.domain.support

import java.time.LocalDate

data class SupportProgram(
    val id: String,
    val title: String,
    val organization: String,
    val summary: String,
    val categories: List<String>,
    val regions: List<String>,
    val targetDescription: String,
    val supportAmount: String,
    val applicationPeriod: String,
    val applicationStartDate: LocalDate?,
    val applicationEndDate: LocalDate?,
    val status: SupportProgramStatus,
    val sourceName: String,
    val sourceUrl: String,
    val matchedReasons: List<String>,
)
